rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_service_info.hpp

```diff
@@ -159,4 +169,49 @@ class ServiceListener : public eprosima::fastrtps::SubscriberListener
   std::condition_variable * conditionVariable_ RCPPUTILS_TSA_GUARDED_BY(internalMutex_);
 };
 
+class ServicePubListener : public eprosima::fastrtps::PublisherListener
+{
+public:
+  ServicePubListener() = default;
+
+  template<class Rep, class Period>
+  bool wait_for_subscription(
+    const eprosima::fastrtps::rtps::GUID_t & guid,
+    const std::chrono::duration<Rep, Period> & rel_time)
+  {
+    auto guid_is_present = [this, guid]() -> bool
+      {
+        return subscriptions_.find(guid) != subscriptions_.end();
```

Comment:
`clang` warns about this line: https://ci.ros2.org/view/nightly/job/nightly_linux_clang_libcxx/483/gcc/folder.862079900/

> reading variable 'subscriptions_' requires holding mutex 'mutex_' [-Wthread-safety-analysis] return subscriptions_.find(guid) != subscriptions_.end();